digest/src/core_api/rt_variable.rs

```diff
@@ -118,6 +135,67 @@ where
     }
 }
 
+impl<T> SerializableState for RtVariableCoreWrapper<T>
+where
+    T: VariableOutputCore + UpdateCore + SerializableState,
+    T::BlockSize: IsLess<U256>,
+    Le<T::BlockSize, U256>: NonZero,
+    T::SerializedStateSize: Add<U1>,
+    Sum<T::SerializedStateSize, U1>: Add<T::BlockSize> + ArraySize,
+    Sum<Sum<T::SerializedStateSize, U1>, T::BlockSize>: Add<U1> + ArraySize,
+    Sum<Sum<Sum<T::SerializedStateSize, U1>, T::BlockSize>, U1>:
+        Sub<T::SerializedStateSize> + ArraySize,
+    Diff<Sum<Sum<Sum<T::SerializedStateSize, U1>, T::BlockSize>, U1>, T::SerializedStateSize>:
+        Sub<U1> + ArraySize,
+    Diff<
+        Diff<Sum<Sum<Sum<T::SerializedStateSize, U1>, T::BlockSize>, U1>, T::SerializedStateSize>,
+        U1,
+    >: Sub<T::BlockSize> + ArraySize,
+    Diff<
+        Diff<
+            Diff<
+                Sum<Sum<Sum<T::SerializedStateSize, U1>, T::BlockSize>, U1>,
+                T::SerializedStateSize,
+            >,
+            U1,
+        >,
+        T::BlockSize,
+    >: ArraySize,
+{
+    type SerializedStateSize = Sum<Sum<Sum<T::SerializedStateSize, U1>, T::BlockSize>, U1>;
```

Comment:
Curious if e.g. the `Sum`-computed types might be clearer with type aliases, similar to something like https://github.com/RustCrypto/signatures/blob/7bf1313/ecdsa/src/der.rs#L45

Reply:
```
type Sum1<T> = Sum<T, U1>;
type Diff1<T> = Diff<T, U1>;
type SumBlockSize<T, B> = Sum<T, <B as BlockSizeUser>::BlockSize>;
type DiffBlockSize<T, B> = Diff<T, <B as BlockSizeUser>::BlockSize>;
type RtVariableCoreWrapperSerializedStateSize<T> =
    Sum1<SumBlockSize<Sum1<<T as SerializableState>::SerializedStateSize>, T>>;

impl<T> SerializableState for RtVariableCoreWrapper<T>
where
    T: VariableOutputCore + UpdateCore + SerializableState,
    T::BlockSize: IsLess<U256>,
    Le<T::BlockSize, U256>: NonZero,
    T::SerializedStateSize: Add<U1>,
    Sum1<T::SerializedStateSize>: Add<T::BlockSize> + ArraySize,
    SumBlockSize<Sum1<T::SerializedStateSize>, T>: Add<U1> + ArraySize,
    RtVariableCoreWrapperSerializedStateSize<T>: Sub<T::SerializedStateSize> + ArraySize,
    Diff<RtVariableCoreWrapperSerializedStateSize<T>, T::SerializedStateSize>: Sub<U1> + ArraySize,
    Diff1<Diff<RtVariableCoreWrapperSerializedStateSize<T>, T::SerializedStateSize>>:
        Sub<T::BlockSize> + ArraySize,
    DiffBlockSize<
        Diff1<Diff<RtVariableCoreWrapperSerializedStateSize<T>, T::SerializedStateSize>>,
        T,
    >: ArraySize,
{
    type SerializedStateSize = RtVariableCoreWrapperSerializedStateSize<T>;
```

What do you think about this one?
But `Sum1`, `Diff1`, `SumBlockSize` and `DiffBlockSize` should go to the `typenum` crate. Or to the `crypto-common`.
Also, something like `SumSerializedState<T, S>` and `DiffSerializedState<T, S>` is possible.

Reply:
Not sure `Sum1` and `Diff1` really make anything clearer, but the others look good

Reply:
Done.